From 296e9ffb93c39f4498b268c6f54323707b5c3775 Mon Sep 17 00:00:00 2001
From: maxwen <max.weninger@gmail.com>
Date: Sun, 4 Oct 2026 10:39:20 +0700
Subject: [PATCH] Settings: allow disable of screenshot shutter sound [2/2]

Change-Id: I159585ff4ed8b1c7d45613dc5d39309b5eeb0ea3
(cherry picked from commit 805d2149aba9ba9d1d3a29dad43b512701f78c86)

Adapt to Lineage:
- Use lineageos.preference.LineageSystemSettingSwitchPreference
  instead of org.evolution.settings.preferences.SystemSettingSwitchPreference
- Put string in res/values/strings.xml instead of evolution_strings.xml
- defaultValue 1 -> true

Change-Id: I69ee5567e65d888a2ebde111cec942b53e563bcf
---
 res/values/strings.xml                        |  3 ++
 res/xml/sound_settings.xml                    |  6 ++++
 .../ScreenshotSoundPreferenceController.java  | 36 +++++++++++++++++++
 .../settings/notification/SoundSettings.java  |  3 ++
 4 files changed, 48 insertions(+)
 create mode 100644 src/com/android/settings/notification/ScreenshotSoundPreferenceController.java

diff --git a/res/values/strings.xml b/res/values/strings.xml
index a84186b..ac692e4 100644
--- a/res/values/strings.xml
+++ b/res/values/strings.xml
@@ -9078,6 +9078,9 @@
     <!-- Sound: Other sounds: Title for the option enabling touch sounds. [CHAR LIMIT=30] -->
     <string name="touch_sounds_title">Tap &amp; click sounds</string>
 
+    <!-- Sound: Other sounds: Title for the option enabling screenshot sound. -->
+    <string name="screenshot_shutter_sound_title">Screenshot sound</string>
+
     <!-- Sound: Other sounds: Title for the option enabling the vibrate icon. [CHAR LIMIT=50] -->
     <string name="vibrate_icon_title">Always show icon when in vibrate mode</string>
 
diff --git a/res/xml/sound_settings.xml b/res/xml/sound_settings.xml
index 2ddbd52..7b73639 100644
--- a/res/xml/sound_settings.xml
+++ b/res/xml/sound_settings.xml
@@ -230,6 +230,12 @@
             android:title="@string/touch_sounds_title"
             android:order="-30"/>
 
+        <!-- Screenshot sound -->
+        <SwitchPreferenceCompat
+            android:key="screenshot_shutter_sound"
+            android:title="@string/screenshot_shutter_sound_title"
+            android:order="-28" />
+
         <!-- Show vibrate icon in status bar -->
         <SwitchPreferenceCompat
             android:key="vibrate_icon"
diff --git a/src/com/android/settings/notification/ScreenshotSoundPreferenceController.java b/src/com/android/settings/notification/ScreenshotSoundPreferenceController.java
new file mode 100644
index 0000000..eed56d2
--- /dev/null
+++ b/src/com/android/settings/notification/ScreenshotSoundPreferenceController.java
@@ -0,0 +1,36 @@
+/*
+ * Copyright (C) 2026 The ExtraParts Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.android.settings.notification;
+
+import static com.android.settings.notification.SettingPref.TYPE_SYSTEM;
+
+import android.content.Context;
+
+import com.android.settings.SettingsPreferenceFragment;
+import com.android.settingslib.core.lifecycle.Lifecycle;
+
+public class ScreenshotSoundPreferenceController extends SettingPrefController {
+
+    private static final String KEY_SCREENSHOT_SHUTTER_SOUND = "screenshot_shutter_sound";
+
+    public ScreenshotSoundPreferenceController(Context context, SettingsPreferenceFragment parent,
+            Lifecycle lifecycle) {
+        super(context, parent, lifecycle);
+        mPreference = new SettingPref(
+            TYPE_SYSTEM, KEY_SCREENSHOT_SHUTTER_SOUND, KEY_SCREENSHOT_SHUTTER_SOUND, DEFAULT_ON);
+    }
+}
diff --git a/src/com/android/settings/notification/SoundSettings.java b/src/com/android/settings/notification/SoundSettings.java
index e0da708..521c5f7 100644
--- a/src/com/android/settings/notification/SoundSettings.java
+++ b/src/com/android/settings/notification/SoundSettings.java
@@ -272,6 +272,8 @@ public class SoundSettings extends DashboardFragment implements OnActivityResult
                 new DockingSoundPreferenceController(context, fragment, lifecycle);
         final TouchSoundPreferenceController touchSoundPreferenceController =
                 new TouchSoundPreferenceController(context, fragment, lifecycle);
+        final ScreenshotSoundPreferenceController screenshotSoundPreferenceController =
+                new ScreenshotSoundPreferenceController(context, fragment, lifecycle);
         final DockAudioMediaPreferenceController dockAudioMediaPreferenceController =
                 new DockAudioMediaPreferenceController(context, fragment, lifecycle);
         final BootSoundPreferenceController bootSoundPreferenceController =
@@ -286,6 +288,7 @@ public class SoundSettings extends DashboardFragment implements OnActivityResult
         controllers.add(chargingSoundPreferenceController);
         controllers.add(dockingSoundPreferenceController);
         controllers.add(touchSoundPreferenceController);
+        controllers.add(screenshotSoundPreferenceController);
         controllers.add(vibrateIconPreferenceController);
         controllers.add(dockAudioMediaPreferenceController);
         controllers.add(bootSoundPreferenceController);
-- 
2.54.0

